akrivis-ingestor/src/main/java/org/kie/akrivis/scheduler/JobExecutor.java

```diff
@@ -20,18 +24,39 @@ public class JobExecutor {
 
     @Transactional
     public RawData run(Long jobId, IngestorHttpClient client) {
-        Job jobPersisted = jobRepository.findById(jobId);
+        final Job jobPersisted = jobRepository.findById(jobId);
         jobPersisted.lastRun = Instant.now();
         LOG.info("Executing job: %d at %s".formatted(jobPersisted.id, jobPersisted.lastRun.toString()));
 
-        RawData newRawData = new RawData();
+        final Optional<RawData> latest = jobRepository.findLatestRawDataByJobId(jobPersisted.id);
+        final String data = client.fetchData(jobPersisted.endpoint);
 
-        newRawData.data =  client.fetchData(jobPersisted.endpoint);
+        if (latest.isPresent() && jsonEquals(data, latest.get().data)) {
```

Comment:
I think the biggest problem here is that if the payload is identical, it doesn't do anything (except for logging in the logger). 

If we want to have this behaviour, to avoid having the same payload stored multiple times, we want to make sure that there's a ts field in the job that states when was the last execution of the fetching, so that we're sure that it's still fetching.

Reply:
When the job is persisted under that if-clause it updates the "lastRun" field of the Job for the database. That does the minimal of notifying we fetched and this is when, but the data is still the same. I think that is what you are saying, but maybe you want more.

On the long run this needs to be improved. Since now there is no way to know if fetching failed and why. The connection and credential issues should be logged and if we provide a log of runs showing how often the data changes then the user can adjust the polling rate to match that. Likely this means a new table where the fetches are stored.